lib/omnibus/project.rb

```diff
@@ -1072,6 +1072,24 @@ def extra_package_files(val = NULL)
       @extra_package_files ||= []
     end
 
+    #
+    # The list of files in Windows build whose symbol need to be stripped.
+    #
```

Comment:
Since we do expect values to be passed in parameters, and the method can raise in case of an invalid parameter, let's add `@param`, `@example` and`@raise` sections to the method documentation:

```
    # @example
    #   windows_files_to_strip [
    #     "/path/to/binary/a.exe",
    #     "/path/to/binary/b.exe"
    #   ]
    # @param [Array<String>] val
    #   the list to files to strip
    #
    # @raise [InvalidValue] if the provided value is not an array
    #
```

---------

Side note: the usual way to store an array for a`Project` or `Software` object and allow the project / software definition file to add items to it in omnibus is to have:
- a DSL method `windows_file_to_strip` that takes one string, and adds it to the `windows_files_to_strip` array, which is exposed.
- another method `windows_files_to_strip` that returns the instance object `@windows_files_to_strip` (initialized to `[]` if empty).
- then, in the project file, do:
```
windows_file_to_strip "/path/to/binary/a.exe"
windows_file_to_strip "/path/to/binary/b.exe"
```
See for example the `extra_package_file` and `extra_package_files` methods in `lib/omnibus/project.rb`.

That way you avoid having to do an Array type check, and you also avoid syntax errors in the project definition file (forgetting to add a comma when adding an element to the list, etc.)

Reply:
Thanks for the feedback. I've changed the function following extra_package_files.